proxy/http2/Http2ClientSession.cc

```diff
@@ -116,9 +117,17 @@ Http2ClientSession::new_connection(NetVConnection *new_vc, MIOBuffer *iobuf, IOB
   this->_read_buffer_reader     = reader ? reader : this->read_buffer->alloc_reader();
 
   // This block size is the buffer size that we pass to SSLWriteBuffer
-  auto buffer_block_size_index   = iobuffer_size_to_index(Http2::write_buffer_block_size, MAX_BUFFER_SIZE_INDEX);
-  this->write_buffer             = new_MIOBuffer(buffer_block_size_index);
-  this->write_buffer->water_mark = Http2::buffer_water_mark;
+  auto buffer_block_size_index = iobuffer_size_to_index(Http2::write_buffer_block_size, MAX_BUFFER_SIZE_INDEX);
+  this->write_buffer           = new_MIOBuffer(buffer_block_size_index);
+
+  uint32_t buffer_water_mark;
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(this->_vc);
```

Comment:
Sorry for bugging you on this, but the point of TLSSomethingSupport is to remove dependencies for SSLNetVC. Including P_SSLNetVConnection.h brings a lot of dependencies unnecessarily (I haven't completely removed the dependency so there's no difference at the moment though).

You could include "TLSSNISupport.h" instead, and access the member like:
```
TLSSNISupport *snis = dynamic_cast<TLSSNISupport *>(this->_vc);
if (snis && snis->hints_from_sni....
```

Reply:
I think I finally understand what you're getting at here. Take another look, please. 

I'm sorry for being slow here. I thought it was weird there was no other casts to SSLNetVC.